R/new_matrixfit.R

```diff
@@ -489,8 +489,13 @@ make_parind <- function (parlist, corr_matrix_size, length_t, summands = 1) {
   n_corr <- corr_matrix_size
   ## parind is the index vector for the matrix elements
   parind <- array(1, dim = c(n_corr*length_t, 2))
+  parind_aux <- array(1, dim = c(n_corr*length_t, 2))
   for (i in 1:n_corr) {
     parind[((i-1)*length_t+1):(i*length_t), ] <- t(array(parlist[, i] + 1, dim = c(2, length_t)))
+    parind_aux[((i-1)*length_t+1):(i*length_t), ] <- t(array(parlist[, i] + 1, dim = c(2, length_t)))
```

Comment:
because ```parind``` is grown iteratively below and the original ```parind``` is required. One could of course refer to the first couple of columns of ```parind``` in the loop below for the same effect.